### ISSUE TITLE
Access from a provider or config block not possible with current implementation choices.

### ISSUE THREAD

**Issue by wayofspark** [gsklee/ngStorage#16]
Hi,

Could it be possible to register $sessionStorage and $localStorage using providers instead of factory to allow usage of storage in application providers / configuration blocks.

I know that the usage of $rootScope you do in your implementation will cause problem in this approach, but maybe you could just implement good old plain get() and set() methods to access the local/session storage from providers without providing the angular binding magic ?

My use case is simple (and I think, one of the most commun use case for local/session storage ) : a configuration block who must read in the sessionStorage to inject a custom security header using $httpProvider.

**Comment by SirDarquan** [comment]
I know this is an old thread but I just encountered this exact issue and would like to add my two cents, if I may.

In my case, I'm using angular-translate to determine the preferred language of the users browser. angular-translate has a provider that exposes a 'determinePreferredLanguage' function that can be customized. It's at this point that I need to access the LocalStorage but since I'm in a provider configuration block, injecting ngStorage isn't possible. I then must use window.localStorage["ngStorage-key"] to retrieve my data which, of course, makes this slightly less testable.

I have a work around in place using the window object and I'm OK with it but a more elegant solution would be better, if possible.

**Comment by tusharmath** [comment]
@SirDarquan  exact same issue. +1

**Comment by brutto** [comment]
+1 to access provider in .config() section

**Comment by cdparra** [comment]
+1, facing the exact same issue, which is making me reconsider the other localstorage options

**Comment by egilkh** [comment]
Anyone willing to test using the ngStorage.js in PR https://github.com/gsklee/ngStorage/issues/138 ?

**Comment by SirDarquan** [comment]
I wouldn't mind testing it but I'm still somewhat of a git/github noob so I don't get how to actually pull it down using bower. If you can point me in the right direction...

**Comment by SirDarquan** [comment]
I decided to just paste the source into a file and reference it that way. Anyway, setup was straight-forward: the $localStorage service has a $localStorageProvider and the same with $sessionStorage service. What I didn't get immediately was that for the provider, we MUST use the get and setter provided to retrieve the keys from storage. Make sure this is explained in the read me for future reference. I initially thought that somehow the provider worked the same way as the service similar to how the ui-router project does it. Other than that, I think it's great. Works perfectly. 

I also noticed the addition of the setKeyPrefix method which is also awesome and seems to work but that feels like feature creep for this particular issue. Great job though. Can't wait until it's in the main code!
